ambry-store/src/main/java/com.github.ambry.store/HardDeleter.java

```diff
@@ -234,7 +234,11 @@ void performRecovery() throws StoreException {
       }
     } catch (IOException e) {
       metrics.hardDeleteExceptionsCount.inc();
-      throw new StoreException("IO exception while performing hard delete ", e, StoreErrorCodes.IOError);
+      if (e.getClass() == IOException.class && e.getMessage().equals(StoreException.IO_ERROR_STR)) {
```

Comment:
first condition is always true (same in other parts of this class)

Reply:
But does it really matter? The second condition won't succeed for those exceptions anyway. I'm just saying that the first condition is not necessary for correctness